### PR TITLE
First week of Member role

### PR DESCRIPTION
**Pull request by DenisCo1100100**
Added tables: UserTask, TaskTrack, TaskState;
Procedures: SetUserTaskAsFail, SetUserTaskAsSuccess;
View: vUserTask;

### CODE REVIEW

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskState.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿CREATE TABLE [dbo].[TaskState]
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/TaskState.sql:1]: Make all tables in plural form

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿CREATE TABLE [dbo].[TaskTrack]
+(
+	[TaskTrackId] INT IDENTITY(1,1) NOT NULL,
+	[UserTaskId] INT NOT NULL,
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql:4]: constaint?

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿CREATE TABLE [dbo].[UserTask]
+(
+	[UserTaskId] INT IDENTITY(1,1) NOT NULL,
+	[TaskId] INT NOT NULL,
+	[UserId] INT NOT NULL,
+	[StateId] INT NOT NULL,
+
+	CONSTRAINT [PK_UserTask] PRIMARY KEY CLUSTERED([UserTaskId] ASC),
+
+	CONSTRAINT [FK_UserTask_TaskTrack] FOREIGN KEY ([TaskId]) REFERENCES [TaskTrack](TaskTrackId)
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql:10]: Wait what? How can you join UserTasks and TaskTracks tables?

File: DIMS-Core/DIMS-Core/appsettings.json [hunk 4]
```diff
@@ -8,5 +8,8 @@
             }
         }
     },
-    "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "ConnectionStrings": {
+    "DIMSDatabase": "Data Source=home-pc;Initial Catalog=DIMSDatabase;Integrated Security=True;"
+  }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/appsettings.json:14]: ??

**Review by PushMeNow — CHANGES_REQUESTED** [review]

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskState.sql (hunk 1 above)
Comment (DenisCo1100100) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/TaskState.sql]: fixed

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql (hunk 2 above)
Comment (DenisCo1100100) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql]: fixed

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql (hunk 3 above)
Comment (DenisCo1100100) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql]: fixed

File: DIMS-Core/DIMS-Core/appsettings.json (hunk 4 above)
Comment (DenisCo1100100) [reply DIMS-Core/DIMS-Core/appsettings.json]: fixed

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql (hunk 3 above)
Comment (PushMeNow) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql]: Still not fixed.

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserTasks.sql [hunk 5]
```diff
@@ -0,0 +1,21 @@
+﻿CREATE TABLE [dbo].[UserTasks]
+( 
+	[UserTaskId] INT IDENTITY(1,1) NOT NULL,
+	[TaskId] INT NOT NULL,
+	[UserId] INT NOT NULL,
+	[StateId] INT NOT NULL,
+
+	CONSTRAINT [PK_UserTasks] PRIMARY KEY CLUSTERED([UserTaskId] ASC),
+
+	FOREIGN KEY([TaskId]) REFERENCES [dbo].[TaskTracks] ([TaskTrackId])
```
Comment (PushMeNow) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/UserTasks.sql:10]: Still not fixed

**Review by PushMeNow — CHANGES_REQUESTED** [review]

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserTasks.sql (hunk 5 above)
Comment (DenisCo1100100) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserTasks.sql]: Fixed

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql [hunk 6]
```diff
@@ -0,0 +1,7 @@
+﻿CREATE VIEW [vUserTask]
+AS SELECT [UserTask].UserId, [UserTask].TaskId, [Task].Name AS TaskName,
+		  [Task].Description AS Descriptions,
+		  [Task].StartDate, [Task].DeadlineDate, [TaskState].StateName
+FROM	[Task] INNER JOIN
+		[UserTask] ON [Task].TaskId = [UserTask].UserTaskId INNER JOIN
+		[TaskState] ON [UserTask].StateId = [TaskState].StateId
```
Comment (kuzmiich) [inline DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql:7]: prettify this statement, pls.

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTracks.sql [hunk 7]
```diff
@@ -0,0 +1,11 @@
+﻿CREATE TABLE [dbo].[TaskTracks]
+(
+	[TaskTrackId] INT IDENTITY(1,1) NOT NULL,
+	[UserTaskId] INT NOT NULL,
+	[TrackDate] DATE NOT NULL,
+	[TrackNote] NVARCHAR(50) NOT NULL,
+
+	CONSTRAINT [PK_TaskTracks] PRIMARY KEY CLUSTERED([TaskTrackId] ASC),
+	
+	FOREIGN KEY([UserTaskId]) REFERENCES [dbo].[UserTasks]([UserTaskId])
```
Comment (kuzmiich) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTracks.sql:10]: Should you delete cascade UserTask?

Comment (DenisCo1100100) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTracks.sql:10]: Fixed

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql (hunk 6 above)
Comment (DenisCo1100100) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql]: Fixed

**Review by kuzmiich — APPROVED** [review]

**Review by PushMeNow — APPROVED** [review]

**Pull request merged**